internal/controllers/clusterextension_controller.go

```diff
@@ -304,12 +321,36 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 		},
 	}
 
-	rel, state, err := r.getReleaseState(ac, ext, chrt, values, post)
+	rel, desiredRel, state, err := r.getReleaseState(ac, ext, chrt, values, post)
 	if err != nil {
 		setInstalledStatusConditionFailed(ext, fmt.Sprintf("%s:%v", ocv1alpha1.ReasonErrorGettingReleaseState, err))
 		return ctrl.Result{}, err
 	}
 
+	for _, preflight := range r.Preflights {
+		if ext.Spec.Preflight != nil {
+			if _, ok := preflight.(*crdupgradesafety.Preflight); ok && ext.Spec.Preflight.CRDUpgradeSafety.Disabled {
```

Comment:
Even though it is a required field, the `ext.Spec.Preflight.CRDUpgradeSafety` is a pointer right? Should we add a nil check for this field as well as an extra safeguard to ensure we don't hit a nil pointer error on that field?

Reply:
That's a good idea, updated in [latest commit](https://github.com/operator-framework/operator-controller/pull/979/commits/06914d241169413f7cd299dba11e3208328792bd)